gtfs/spec/en/reference.md

```diff
@@ -472,8 +480,8 @@ Assigns stops from [stops.txt](#stopstxt) to areas.
 
 |  Field Name | Type | Presence | Description |
 |  ------ | ------ | ------ | ------ |
-| `area_id` | Foreign ID referencing `areas.area_id` | **Required** | Identifies an area to which one or multiple `stop_id`s belong. The same `stop_id` may be defined in many `area_id`s. |
-| `stop_id` | Foreign ID referencing `stops.stop_id` | **Required** | Identifies a stop. If a station (i.e. a stop with `stops.location_type=1`) is defined in this field, it is assumed that all of its platforms (i.e. all stops with `stops.location_type=0` that have this station defined as `stops.parent_station`) are part of the same area. This behavior can be overridden by assigning platforms to other areas. |
+| `area_id` | Foreign ID referencing `areas.area_id` | **Required** | Identifies an area to which one or multiple `stop_id`s belong. The same `stop_id` may be defined in many `area_id`s.<br><br>May also identify a group of stops and/or GeoJSON locations that together indicate locations where a rider may request pickup or drop off.<br><br>It is forbidden to define an `area_id` with the same value as a `stop_id` or `id` from `locations.geojson`. |
```

Comment:
I understand the need to have distinct values between area_id, stop_id, geojson.id, but this is not something managable (even validation is not simple, but to do this in the DB level is very hard if not impossible)

I'm not sure how can this be improved. If the id-s are integers then it's not really possible. But if they are strings then maybe a good practice could be to use some prefix, ie: for area_id: a_123, for stop_id: s_123, for geojson_id: g_123.

Reply:
We've managed to keep these values distinct at the DB level with no issue. Our `area_id`s are generated in the same table as our `stop_id`s in ascending order, so there's no risk of overlap there, and we do use a prefix for location `id`s.

Using a prefix to ensure these values don't overlap is good advice, but I do not think it needs to be a requirement of the spec.

Logically, I guess I don't see the validation hurdle: "Does this id appear in stops.txt? Does this id appear in locations.geojson? Does this id appear in areas.txt? If no to all, passes validation" seems fairly straightforward.

Reply:
As a consumer (OBA, OTP) I've also had no issue with these IDs sharing a global ID space. Yes, it does make validation and lookup a bit more complicated but considering all the other complexities of the Flex spec, this is the easy part.

It's probably a good idea to prefix your IDs but that should really be left to the producer. I would not want to add any kind of ID requirements to the spec. IMO GTFS should treat them as transparent identifiers with no meta data encoded into them, which it currently does.